packages/runner/src/recipe-manager.ts

```diff
@@ -11,6 +12,18 @@ export const recipeMetaSchema = {
     spec: { type: "string" },
     parents: { type: "array", items: { type: "string" } },
     recipeName: { type: "string" },
+    additionalSources: { 
```

Comment:
Sketching this out, I replaced `src` with multiple sources (as this needs a list of sources with their names, as well as the name of the entry, names are "/paths/like/this", maybe we can bless a default `index.ts` or something in the future) -- reasoning here is we can represent a single file recipe as a multi file one rather than having an explicit "single file only" execution branch e.g. `{ entry: "/index.ts", sources: { "/index.ts": "import { .. } ..." }}`, though probably will need to cast from the JSONSchema form to what Engine wants (we represent sources as both `Record<string, string>` and `{ name: string, contents: string }[]` in different places), though a simple cast

Reply:
I shared this in discord, adding comment here for future us:

the reason I kept `src` and stored the "additional" source files in a new array is so that existing charms/recipes/spaces would continue to work.

If we want to make this a breaking change we can be more aggressive with naming